cluster/sbin/snappy-start-all.sh

```diff
@@ -32,23 +32,38 @@ if [ -f "${MEMBERS_FILE}" ]; then
   rm $MEMBERS_FILE
 fi
 
-# Check for foreground start
 BACKGROUND=-bg
-if [ "$1" = "-bg" -o "$1" = "--background" ]; then
-  shift
-fi
-if [ "$1" = "-fg" -o "$1" = "--foreground" ]; then
-  BACKGROUND=""
+clustermode=
+
+while (( "$#" )); do
+  param="$1"
+  case $param in
+    -bg | --background)
+      # Check for background start
+      BACKGROUND=-bg
+    ;;
+    -fg | --foreground)
+      # Check for foreground start
+      BACKGROUND=""
+    ;;
+    rowstore)
+      clustermode="rowstore"
+    ;;
+    *)
+      echo "Invalid option: $param"
```

Comment:
You care for this echo? I think later somewhere down it will complain. Anyways no harm.

Reply:
Isn't this more standard:

while test $# -gt 0
do
     param="$1"
     shift
done
}